### PR TITLE
ocamlPackages.dbf: init at 1.1.0

### PR DESCRIPTION
**Pull request by tournemire**
Dbf is a small OCaml library meant to parse [dBase](https://en.wikipedia.org/wiki/DBase) database files.
 
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/dbf/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, buildDunePackage, fetchFromGitHub, ocaml, ppx_cstruct, rresult
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/dbf/default.nix:1]:
```suggestion
{ lib, buildDunePackage, fetchFromGitHub, ppx_cstruct, rresult
```

File: pkgs/development/ocaml-modules/dbf/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+, cstruct-unix, core_kernel }:
+
+buildDunePackage rec {
+  pname = "dbf";
+  version = "0.1.1";
+
+  minimumOCamlVersion = "4.02";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/dbf/default.nix:8]:
```suggestion
  minimalOCamlVersion = "4.02";
```

File: pkgs/top-level/ocaml-packages.nix [hunk 3]
```diff
@@ -252,6 +252,8 @@ let
 
     dap =  callPackage ../development/ocaml-modules/dap { };
 
+    dbf =  callPackage ../development/ocaml-modules/dbf { };
+
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix:256]: Please keep the attributes (locally) sorted in alphabetical order.

File: pkgs/development/ocaml-modules/dbf/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  useDune2 = true;
+
+  src = fetchFromGitHub {
+    owner = "pveber";
+    repo = "dbf";
+    rev = "${version}";
+    sha256 = "sha256-h1K5YDLbXGEJi/quKXvSR0gZ+WkBzut7AsVFv+Bm8/g=";
+  };
+
+  propagatedBuildInputs = [ ppx_cstruct rresult cstruct-unix core_kernel ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/dbf/default.nix:19]: I think that `ppx_cstruct` need not be propagated.

File: pkgs/development/ocaml-modules/dbf/default.nix (hunk 2 above)
Comment (tournemire) [reply pkgs/development/ocaml-modules/dbf/default.nix]: The nixpkgs manual indeed uses "minimal" but I also see "minimum" in many OCaml packages. How do these differ ? buildDunePackage seems to use either indiscriminately.

**Comment by tournemire** [comment]
Applied suggested changes and fixed minimal OCaml version, which I initially forgot to test.

File: pkgs/development/ocaml-modules/dbf/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/dbf/default.nix]: “minimum” works but 1. it is a grammatical error and 2. it changes the derivation hash (which is wrong at it is only meta-data). Only the correct spelling is documented.

**Pull request merged**